### PR TITLE
docs(bastion): improve how to login into servers

### PR DESCRIPTION
**Pull request by MarcoIeni**

### CODE REVIEW

File: src/infra/docs/bastion.md [hunk 1]
```diff
@@ -12,20 +12,22 @@ directly to a production server with SSH. Instead, all connections must come
...
-```
-Host servername.infra.rust-lang.org
-    ProxyJump bastion.infra.rust-lang.org
-```
+To log into a server through the bastion, use one of the following methods:
+
+- use SSH's `-J` flag:
+  ```
+  ssh -J <username>@bastion.infra.rust-lang.org <username>@servername.infra.rust-lang.org
+  ```
```
Comment (jdno) [inline src/infra/docs/bastion.md:20]: Nit: I would follow the convention in other parts of the document to surround code blocks with empty lines. Some for the other code blocks a bit further down...

File: src/infra/docs/bastion.md [hunk 2]
```diff
@@ -12,20 +12,22 @@ directly to a production server with SSH. Instead, all connections must come
...
-connecting to a host. Add this snippet to your SSH configuration file (usually
-located in `~/.ssh/config`):
-
-```
-Host servername.infra.rust-lang.org
-    ProxyJump bastion.infra.rust-lang.org
-```
+To log into a server through the bastion, use one of the following methods:
+
+- use SSH's `-J` flag:
```
Comment (jdno) [inline src/infra/docs/bastion.md:17]: Nit: Let's capitalize the first letter in lists to ensure consistency

```suggestion
- Use SSH's `-J` flag:
```

File: src/infra/docs/bastion.md [hunk 3]
```diff
@@ -12,20 +12,22 @@ directly to a production server with SSH. Instead, all connections must come
...
+  ```
+  ssh -J <username>@bastion.infra.rust-lang.org <username>@servername.infra.rust-lang.org
+  ```
+- configure your SSH client to always jump through the bastion when connecting to a host:
```
Comment (jdno) [inline src/infra/docs/bastion.md:21]: See above

```suggestion
- Configure your SSH client to always jump through the bastion when connecting to a host:
```

File: src/infra/docs/bastion.md [hunk 4]
```diff
@@ -12,20 +12,22 @@ directly to a production server with SSH. Instead, all connections must come
...
+  - Add this snippet to your SSH configuration file (usually located in `~/.ssh/config`):
+    ```
+    Host servername.infra.rust-lang.org
+        ProxyJump <username>@bastion.infra.rust-lang.org
+    ```
+  - use SSH:
```
Comment (jdno) [inline src/infra/docs/bastion.md:27]: See above

```suggestion
  - Use SSH:
```

**Review by jdno — APPROVED** [review]
Small suggestions to ensure a consistent style, but the content is good. :+1:

File: src/infra/docs/bastion.md (hunk 1 above)
Comment (MarcoIeni) [reply src/infra/docs/bastion.md:21]: I formatted the entire document with prettier 👍

**Review by jdno — APPROVED** [review]
:shipit:

**Pull request merged**